### PR TITLE
Fix version_parse on windows

### PR DESCRIPTION
**Pull request by JLockerman**
`snprint` has different behavior on some windows versions than on unix.
This commit replaces the `snprintf` version with one that uses `sscanf`,
which behaves the same on all platforms. (This is a higher priority
commit than it would appear, since we're tracking down a stack-overflow,
and this is one of the candidates for the cause)

Requesting reviews by @amytai since she owns the version-checking code and @erimatnor since he wrote the last version of the version_parse code.

### CODE REVIEW

File: src/version.c [hunk 1]
```diff
@@ -74,60 +74,40 @@ version_cmp(VersionInfo *v1, VersionInfo *v2)
...
+	 *
+	 * after parsing each field in the version-string, we output the number of
+	 * bytes currently parsed, a version-string is valid if all of the bytes
+	 * of the string were parsable using the above grammar.
+	 */
+	fields_parsed = sscanf(version, "%lu%n.%lu%n.%lu%n-%" STR(VERSION_INFO_LEN) "s%n",
+						   &result->version[0], &parsed_len,
+						   &result->version[1], &parsed_len,
+						   &result->version[2], &parsed_len,
+						   result->version_mod, &parsed_len);
```
Comment (cevian) [inline src/version.c]: Please double check if passing `parsed_len` multiple times to sscanf is safe.

Comment (JLockerman) [reply src/version.c]: the standard is ambiguous switched to multiple lengths just to be sure.

**Comment by amytai** [comment]
@erimatnor We might try to release 1.0.1 early next week, which should have these version-parsing/checking changes, so we are merging now.

File: src/version.c [hunk 2]
```diff
@@ -74,60 +74,51 @@ version_cmp(VersionInfo *v1, VersionInfo *v2)
...
+						   &result->version[1], &parse_length[1],
+						   &result->version[2], &parse_length[2],
+						   result->version_mod, &parse_length[3]);
 
-	return true;
+	/*
+	 * sscanf is allowed to return EOF if no parses suceed, so make sure
+	 * fields_parsed is between 1 and 4;
+	 */
+	if(fields_parsed <= 0 || fields_parsed > 4)
```
[the remark this answers is not in the document]
Comment (JLockerman) [reply src/version.c]: Yup, I need to run `pgindent`

**Comment by JLockerman** [comment]
> Also, I never understood from the description how snprintf differs...

**Comment by amytai** [comment]
Just a quick note, @erimatnor. I have a PR ready to go that will move version-checking to the server-side. Unfortunately, however, we still need this version-parsing code as a brief verification that the server response is well-formed, before we print it in the logs.

In other words, if the server tells us that the local version is "out-of-date", then we will want to print the string of the most "up-to-date" version, as returned to us by the server. But before we print this string, we verify that it is "well-formed" using this logic.

**Pull request merged**